### PR TITLE
FAQ: include short answers before linking to longer texts

### PR DESCRIPTION
**Pull request by SethTisue**
sequel to #1848. review by @BalmungSan?

### CODE REVIEW

File: _overviews/FAQ/index.md [hunk 1]
```diff
@@ -110,28 +110,56 @@ One would hope so, but doing it that way was tried and it proved
...
+Types are primarily a compile-time concept. At compile time,
+every expression is assigned a type by the compiler.
+
+Classes are primarily a runtime concept and are platform-dependent.
+At runtime on the JVM, every value is either a primitive value
+or an instance of exactly one class.
+
+Some type information exists only at compile time, for multiple
+reasons, most notoriously [type
+erasure](https://en.wikipedia.org/wiki/Type_erasure).
```
Comment (BalmungSan) [inline _overviews/FAQ/index.md:138]: Big +1 on this, probably on the very first JVM things I learned and it was due **Scala**.

File: _overviews/FAQ/index.md [hunk 2]
```diff
@@ -110,28 +110,56 @@ One would hope so, but doing it that way was tried and it proved
...
-### What does <:< means?
+### What does `<:<` mean?
 
-[blog post: Generalized type constraints](https://blog.bruchez.name/2015/11/generalized-type-constraints-in-scala.html)
+It's a "type constraint", and it comes from the standard library,
+not from the language itself.  See [this blog
+post](https://blog.bruchez.name/2015/11/generalized-type-constraints-in-scala.html).
 
-### Best way to model optional arguments
+### I dislike requiring callers to wrap optional arguments in `Some(...)`; is there a better way?
```
Comment (BalmungSan) [inline _overviews/FAQ/index.md:158]: 😄

File: _overviews/FAQ/index.md [hunk 3]
```diff
@@ -110,28 +110,56 @@ One would hope so, but doing it that way was tried and it proved
...
 
-[answer on Stack Overflow](https://stackoverflow.com/a/65256691/4111404)
+Not really. See [this answer on Stack Overflow](https://stackoverflow.com/a/65256691/4111404).
 
-### Why prfer implicit val over implicit object
+### Why is `implicit val` usually recommended over `implicit object`?
 
-[answer on Stack Overflow](https://stackoverflow.com/a/65258340/4111404)
+The latter has a singleton type, which is too specific.  See [answer on Stack
```
Comment (BalmungSan) [inline _overviews/FAQ/index.md:164]: Maybe a link to something that explains what is a singleton type?
Or maybe that link should be in the SO answer itself?

**Review by BalmungSan — APPROVED** [review]
I really like this, I even thought about doing that in my previous PR. I didn't because the previous ones were only links, which opens the question if we should do the same for the previous ones?

File: _overviews/FAQ/index.md (hunk 3 above)
Comment (SethTisue) [reply _overviews/FAQ/index.md:164]: Sadly the language tour doesn't have a "Singleton" types section. I agree there's room for improvement here, but I'm not going to tackle it right now.

**Comment by SethTisue** [comment]
> I really like this, I even thought about doing that in my previous PR. I didn't because the previous ones were only links, which opens the question if we should do the same for the previous ones?

I think we should. Perhaps I'll get to it in some future PR, and/or perhaps someone else will.

**Pull request merged**